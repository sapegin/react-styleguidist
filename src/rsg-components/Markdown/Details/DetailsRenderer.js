import React from 'react';
import PropTypes from 'prop-types';

import Styled from 'rsg-components/Styled';

const styles = ({ space, color, fontSize, fontFamily }) => ({
	details: {
		fontFamily: fontFamily.base,
		fontSize: fontSize.base,
		color: color.base,
		marginBottom: space[2],
	},
});

export function DetailsRenderer({ classes, children }) {
	return <details className={classes.details}>{children}</details>;
}
DetailsRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.node.isRequired,
};

export default Styled(styles)(DetailsRenderer);
