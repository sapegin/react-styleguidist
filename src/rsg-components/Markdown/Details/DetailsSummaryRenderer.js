import React from 'react';
import PropTypes from 'prop-types';

import Styled from 'rsg-components/Styled';

const styles = ({ space, color, fontSize, fontFamily }) => ({
	summary: {
		fontFamily: fontFamily.base,
		fontSize: fontSize.base,
		color: color.base,
		marginBottom: space[1],
	},
});

export function DetailsSummaryRenderer({ classes, children }) {
	return <summary className={classes.summary}>{children}</summary>;
}
DetailsSummaryRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.node.isRequired,
};

export default Styled(styles)(DetailsSummaryRenderer);
