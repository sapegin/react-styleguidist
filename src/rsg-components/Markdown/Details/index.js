export { default } from 'rsg-components/Markdown/Details/DetailsRenderer';
