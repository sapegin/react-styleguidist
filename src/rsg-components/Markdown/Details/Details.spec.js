import React from 'react';

import Details from './index';

describe('Markdown Details', () => {
	it('should render a Details', () => {
		const actual = render(<Details>This is a hidden text.</Details>);

		expect(actual).toMatchSnapshot();
	});
});
