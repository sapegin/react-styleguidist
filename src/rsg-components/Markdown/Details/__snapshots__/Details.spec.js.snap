// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown Details should render a Details 1`] = `
<details
  class="rsg--details-0"
>
  This is a hidden text.
</details>
`;
